models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql

```diff
@@ -130,48 +125,52 @@ orders_and_trades as (
     inner join {{ source('gnosis_protocol_v2_ethereum', 'GPv2Settlement_call_settle') }}
         on evt_block_number = call_block_number
         and evt_tx_hash = call_tx_hash
--- this is implied by the inner join
---      and call_success = true
 ),
--- Validate Uid <--> app_data mapping here: https://dune.com/queries/1759039?d=1
+
 uid_to_app_id as (
-    select
-        distinct uid,
-        get_json_object(trade, '$.appData') as app_data,
-        get_json_object(trade, '$.receiver') as receiver,
-        get_json_object(trade, '$.sellAmount') as limit_sell_amount,
-        get_json_object(trade, '$.buyAmount') as limit_buy_amount,
-        get_json_object(trade, '$.validTo') as valid_to,
-        get_json_object(trade, '$.flags') as flags
-    from orders_and_trades
-        lateral view posexplode(order_ids) o as i, uid
-        lateral view posexplode(trades) t as j, trade
-    where i = j
+    SELECT
+      uid,
+      from_hex(JSON_EXTRACT_SCALAR(trade, '$.appData')) AS app_data,
+      from_hex(JSON_EXTRACT_SCALAR(trade, '$.receiver')) AS receiver,
+      cast(JSON_EXTRACT_SCALAR(trade, '$.sellAmount') as uint256) AS limit_sell_amount,
+      cast(JSON_EXTRACT_SCALAR(trade, '$.buyAmount') as uint256) AS limit_buy_amount,
+      JSON_EXTRACT_SCALAR(trade, '$.validTo') AS valid_to,
```

Comment:
What is the content of this column? Could it have another data type than varchar?

Reply:
Yea probably. I think it's a Unix time stamp. Will look into it and get it right.

Reply:
Fixed!